tests/callbacks/test_callbacks.py

```diff
@@ -162,23 +221,70 @@ def test_trains(self, cb_cls: type[Callback], device_train_microbatch_size: int,
 
         maybe_patch_context = get_cb_patches(cb_cls)
 
+        # For MLFlowLogger, add a patch for the file_utils.is_directory function
+        if cb_cls.__name__ == 'MLFlowLogger':
+            import unittest.mock as mock
+
+            import mlflow.utils.file_utils
+
+            original_is_directory = mlflow.utils.file_utils.is_directory
+
+            # Create a patched version that returns True for trash directories
+            def patched_is_directory(path):
+                if path.endswith('.trash'):
+                    return True
+                return original_is_directory(path)
+
+            # Apply the patch within the test
+            with mock.patch('mlflow.utils.file_utils.is_directory', patched_is_directory):
+                with maybe_patch_context:
+                    trainer = self._get_trainer(cb, device_train_microbatch_size)
+                    trainer.fit()
+            return  # Exit early as we've already completed the test for MLFlowLogger
+
+        # For all other loggers, proceed normally
         with maybe_patch_context:
             trainer = self._get_trainer(cb, device_train_microbatch_size)
             trainer.fit()
 
     @pytest.mark.filterwarnings('ignore::UserWarning')
     def test_trains_multiple_calls(self, cb_cls: type[Callback], device_train_microbatch_size: int, _remote: bool):
         """
-        Tests that training with multiple fits complete.
-        Note: future functional tests should test for
-        idempotency (e.g functionally)
+        Tests that training with multiple fits complete. Note: future functional tests should test for idempotency (e.g functionally)
         """
         del _remote  # unused. `_remote` must be passed through to parameterize the test markers.
         cb_kwargs = get_cb_kwargs(cb_cls)
         cb = cb_cls(**cb_kwargs)
 
         maybe_patch_context = get_cb_patches(cb_cls)
 
+        # For MLFlowLogger, add a patch for the file_utils.is_directory function
+        if cb_cls.__name__ == 'MLFlowLogger':
```

Comment:
is it possible to centralize this logic (e.g. with `get_cb_patches`)` instead of copying it for each test?